tests/integration-test.js

```diff
@@ -218,6 +218,49 @@ describe('Box Node SDK', function() {
 			});
 	});
 
+	it('should use retry strategy when it is provided and when API calls fail', function() {
+
+		apiMock.get('/2.0/users/me')
+			.reply(500)
+			.get('/2.0/users/me')
+			.reply(502)
+			.get('/2.0/users/me')
+			.reply(429)
+			.get('/2.0/users/me')
+			.reply(500)
+			.get('/2.0/users/me')
+			.reply(500)
+			.get('/2.0/users/me')
+			.reply(200, {
+				type: 'user',
+				id: 'me'
+			});
+
+
+		var sdk = new BoxSDK({
+			clientID: TEST_CLIENT_ID,
+			clientSecret: TEST_CLIENT_SECRET,
+			maxNumRetries: 5,
+			retryIntervalMS: 10,
+			retryStrategy(options) {
```

Comment:
It might be nice to make the `retryStrategy` a `sinon` stub, so that we can assert that it was called the expected number of times as well, and with the correct arguments.